internal/signerverifier/ssh/ssh.go

```diff
@@ -0,0 +1,164 @@
+// SPDX-License-Identifier: Apache-2.0
+package ssh
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"encoding/base64"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/hiddeco/sshsig"
+	"golang.org/x/crypto/ssh"
+)
+
+// FIXME: make this the interface for Keys in TUF metadata
+type Key struct {
+	KeyType string
+	KeyVal  KeyVal
+	Scheme  string
+	keyID   string
```

Comment:
I'm a bit unsure about what to do with keyID. We want the field on the `Key` type, but not (necessarily) in the serialised metadata format. This means we cannot simply use Golang's neat json struct annotations for de/serialization.

There are a few things we could do:
1. include keyid in serialised metadata
   - pro: simple, can use json annotations
   - con:  not TUF spec compliant 
   - con:  duplicates data, as keys in TUF metadata are stored in a `keyid: key` map
2. write custom json parser (go-tuf does this, [for other reasons](https://github.com/theupdateframework/go-tuf/blob/ef106d86c4d7273f82a5ef680d852fb3224cbb10/metadata/marshal.go#L26-L27))
   - pro: TUF spec compliant
   - con: a bit more code
3. compute keyid on-the-fly (go-tuf [seems to do this](https://github.com/theupdateframework/go-tuf/blob/ef106d86c4d7273f82a5ef680d852fb3224cbb10/metadata/keys.go#L121-L133))
   - pros: of 1 and 2
   - con: does not work for arbitrary keyids

Reply:
I think an earlier version of this computed key IDs on the fly when loading it if we didn't already know it. As we usually have `{keyID: key}`, I think we could just set `key.keyID = keyID`? And this should work for arbitrary too? I'm trying to remember why we changed it, it's possible it actually got a bit messy since the GPG case is effectively an arbitrary key ID but we may be able to handle that better now. Let me dig a bit.

Reply:
Is 1 the only option that supports arbitrary key IDs trivially or am I mistaken?

Reply:
Maybe 2 isn't that complex either. I think we just need to implement custom unmarshall functions for the Key containers, and can use json annotations for the rest. I can check out.

Reply:
Sounds good.